src/runtime/manifest-parser.pegjs

```diff
@@ -1614,10 +1614,56 @@ NestedSchemaType = 'inline' whiteSpace? schema:(SchemaInline / TypeName)
     });
   }
 
-Expression
-  = ExpressionEntity 
+QualifiedExpression
+  = FromExpression / WhereExpression / SelectExpression
 
-ExpressionEntity "Expression instantiating a new Arcs entity, e.g. new Foo {x: bar.x}"
+ExpressionWithQualifier
+  = qualifier:QualifiedExpression rest:(multiLineSpace rest:QualifiedExpression)* {
```

Comment:
It seems to me that doing it this way will allow for starting an expression with for example a where clause (or join), I presume we'll need to do validation of that after parsing. I can imagine it's challenging to have the resulting AST coming from the parser have the structure we want.

Reply:
It's possible to do, just tedious in pegjs. I'll give it try.

Reply:
Ok, i restricted it using parser actions.